### PR TITLE
add pre-eip155 evm circuit test

### PR DESCRIPTION
**Pull request by DreamWuGit**
### Description

currently eve circuit lacks pre-eip155 type test, add in this PR and related changes
- [x] get_utils trace support pre-eip155 type tx
- [x] add two tests in begin_tx testing
- [x] adapt MockTransaction to tx with entire signature data.

### CODE REVIEW

File: bus-mapping/src/circuit_input_builder/transaction.rs [hunk 1]
```diff
@@ -343,6 +343,7 @@ impl Transaction {
         );
 
         let tx_type = TxType::get_tx_type(eth_tx);
+        log::info!("bus mapping: tx_type {:?}", tx_type);
```
Comment (lispc) [inline bus-mapping/src/circuit_input_builder/transaction.rs]: change to debug or trace?    "bus mapping" , the module name will be in log line, not needed here.   You can change it like log::debug!("{}th tx, tx hash {}, tx_type {}").

File: geth-utils/l2geth/trace.go [hunk 2]
```diff
@@ -186,6 +198,7 @@ func Trace(config TraceConfig) (*types.BlockTrace, error) {
 	// fmt.Printf("geth-utils: ArchimedesBlock = %d\n", chainConfig.ArchimedesBlock)
 
 	txs := transferTxs(config.Transactions, chainConfig.ChainID)
+	fmt.Printf("geth-utils: transferTxs = %v\n", txs)
```
Comment (lispc) [inline geth-utils/l2geth/trace.go]: really needed?  I noticed the l2geth module does not print other info?

File: mock/src/transaction.rs [hunk 3]
```diff
@@ -508,4 +516,29 @@ impl MockTransaction {
             v
         }
     }
+
+    /// Copy from other MockTransaction instance.
```
Comment (lispc) [inline mock/src/transaction.rs]: why this?  cannot why derive some clone trait/method?

**Comment by lispc** [comment]
leave some small comments

File: bus-mapping/src/circuit_input_builder/transaction.rs (hunk 1 above)
Comment (DreamWuGit) [reply bus-mapping/src/circuit_input_builder/transaction.rs]: changed to  log::debug! in [93967d5](https://github.com/scroll-tech/zkevm-circuits/pull/1412/commits/93967d5fec533fbf2a1b573b87b7dcca4fa38f58)

File: geth-utils/l2geth/trace.go (hunk 2 above)
Comment (DreamWuGit) [reply geth-utils/l2geth/trace.go]: previously for debugging  removed [93967d5](https://github.com/scroll-tech/zkevm-circuits/pull/1412/commits/93967d5fec533fbf2a1b573b87b7dcca4fa38f58)

File: mock/src/transaction.rs (hunk 3 above)
Comment (DreamWuGit) [reply mock/src/transaction.rs]: using this comea from following code encountering rust life time error .
TestContext::new(
            None,
            |accs| {
                accs[0]
                    .address(tx.from.address())
                    .balance(sender_balance)
                    .nonce(tx.nonce);
            },
            |mut txs: Vec<&mut MockTransaction>, _accs| {
                txs[0] = tx.clone(); // will meet life time error 

            },
but now figur out can also use clone_from, changed in  [5e7888b](https://github.com/scroll-tech/zkevm-circuits/pull/1412/commits/5e7888bc2bb67768b8439b35d3e94d95af5ca390).  removed the method.

**Pull request merged**